packages/api/src/external/carequality/gateway/index.ts

```diff
@@ -52,6 +58,38 @@ export async function gatherXCPDGateways(patient: Patient): Promise<Gateways> {
   };
 }
 
+export function facilitiesWithEpicFilter(
+  cqDirectoryEntryModel: CQDirectoryEntry[],
+  isEpicEnabled: boolean
+): CQDirectoryEntry[] {
+  if (isEpicEnabled) {
+    return cqDirectoryEntryModel;
+  }
+
+  return cqDirectoryEntryModel.filter(entry => !isEpicFacility(entry.id, cqDirectoryEntryModel));
+}
+
+function isEpicFacility(gateway: string, cqDirectoryModel: CQDirectoryEntry[]): boolean {
+  const cqDirectoryGW = cqDirectoryModel.find(entry => entry.id === gateway);
```

Comment:
I'd avoid naming it "model" because it gives the sense that operations there are hitting the DB; maybe just `cqDirectoryEntries` since its an array?